assets/js/phoenix_live_view/rendered.js

```diff
@@ -17,6 +19,90 @@ import {
   isCid,
 } from "./utils"
 
+const VOID_TAGS = [
+  "area",
+  "base",
+  "br",
+  "col",
+  "command",
+  "embed",
+  "hr",
+  "img",
+  "input",
+  "keygen",
+  "link",
+  "meta",
+  "param",
+  "source",
+  "track",
+  "wbr"
+]
+
+export let modifyRoot = (html, attrs, clearInnerHTML) => {
+  let i =0
+  let insideComment = false
+  let insideTag = false
+  let tag
+  let beforeTagBuff = []
+  while(i < html.length){
+    let char = html.charAt(i)
+    if(insideComment){
+      if(char === "-" && html.slice(i, i + 3) === "-->"){
+        insideComment = false
+        beforeTagBuff.push("-->")
+        i += 3
+      } else {
+        beforeTagBuff.push(char)
+        i++
+      }
+    } else if(char === "<" && html.slice(i, i + 4) === "<!--"){
+      insideComment = true
+      beforeTagBuff.push("<!--")
+      i += 4
+    } else if(char === "<"){
+      insideTag = true
+      let iAtOpen = i
+      for(i; i < html.length; i++){
+        if([">", " ", "\n", "\t", "\r"].indexOf(html.charAt(i)) >= 0){ break }
+      }
+      tag = html.slice(iAtOpen + 1, i)
+      break
+    } else if(!insideComment && !insideTag){
+      beforeTagBuff.push(char)
+      i++
+    }
+  }
```

Comment:
@chrismccord there's a great deal of complexity here, and if I'm reading it right, it's predominantly for preserving html comments found within tags of arbitrary depth. is that correct?

Reply:
in this case no, it's only about finding the comments before and after a root tag, inject specified attributes, and optionally stripping the innerHTML along with non specified attributes